tools/astarte_dev_tool/lib/commands/system/watch.ex

```diff
@@ -0,0 +1,47 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2024 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule AstarteDevTool.Commands.System.Watch do
+  @moduledoc false
+  use AstarteDevTool.Constants.System
+  require Logger
+  alias AstarteDevTool.Utilities.Process, as: AstarteProcess
+
+  def exec(path) do
+    kill_zombie_process(path)
+
+    with {_result, 0} <-
+           System.cmd(@command, @command_watch_args, @options ++ [cd: path]) do
+      :ok
+    else
+      {:error, reason} -> {:error, "Cannot run system watching: #{reason}"}
+      {result, exit_code} -> {:error, "Cannot exec system.watch: #{result}, #{exit_code}"}
+    end
+  end
+
+  defp kill_zombie_process(path) do
```

Comment:
what if, instead of checking for orphan processes, we set up an internal process listener like https://gitlab.com/Nicd/mbu/-/blob/master/lib/apprunner.exs ?  This one checks for STDIN but can be adapted easily, but it's the idea behind it. A process that kill itself under certain conditions sounds a little bit cleaner than wait for the next run to kill eventual orphans

This want to be a suggestion, not a blocker for the PR, we can discuss this later if you want

Reply:
I'm evaluating to implement it for the next PR (the improvement one)